pkg/liqonet/tunnel/wireguard/driver.go

```diff
@@ -256,11 +268,17 @@ func (w *Wireguard) DisconnectFromEndpoint(tep *netv1alpha1.TunnelEndpoint) erro
 		Peers:        peerCfg,
 	})
 	if err != nil {
-		return fmt.Errorf("failed to remove WireGuard peer with clusterid %s: %w", tep.Spec.ClusterID, err)
+		return fmt.Errorf("failed to remove WireGuard peer with clusterid %s: %w", tep.Spec.ClusterIdentity.ClusterName, err)
 	}
 
-	klog.V(4).Infof("Done removing WireGuard peer with clusterid %s", tep.Spec.ClusterID)
-	delete(w.connections, tep.Spec.ClusterID)
+	remoteKey, err := getKey(tep)
+	if err != nil {
+		return fmt.Errorf("failed to get public key for cluster %s: %w", tep.Spec.ClusterIdentity, err)
+	}
```

Comment:
Should not return an error in case the key is not found, otherwise you might never converge if you end up in a non-standard situation.